### ISSUE TITLE
[Tx signatures] - Invalid signatures of ex owners of a safe

### ISSUE THREAD

**Issue by francovenica** [safe-global/web-core#1463]
## Bug description
A fully signed tx, where one of the signers is now an ex owner, will show it is fully confirmed. 
If any other owner tries to sign to fill the now removed signature from the ex owner an GS026 error pops up

Note: if you are able to re introduce the owner to the safe then its signature is valid again.
Note2: This is only for queued tx.Tx in the history tab (fully executed tx) should show the signatures of old owners just fine

## Environment
 - Browser: Chrome
 - Wallet: MetaMask
 - Chain: Goerli
 - Env: Prod https://app.safe.global/gor:0x24F8542Da0C5f267c177a224270442C73804EE80/home

## Steps to reproduce
 1. Go to safe 2/3 safe at least. Be able to sign with the 3 owners
 2. Create a tx with OwnerA, kicking out ownerB, but do not execute it
 3. Create a send funds tx with OwnerA
 4. Fully sign both tx with OwnerB, but do not execute it yet
 5. Execute the RemoveOwner tx
 6. With ownerC try to sign the Send Funds tx

## Expected result
While the tx is in queue, it should not show the kicked owner as signer anymore
OwnerC should be able to sign the tx normally and execute it

## Obtained result
Even tho the tx shows "1 out of 2" signatures, the "Confirmation" list shows it fully signed 2/2. OwnerC is not able to sign the tx

## Screenshots
![image](https://user-images.githubusercontent.com/4503659/209703141-76d3630a-4024-4d02-9ac3-c0383373bf90.png)
![image](https://user-images.githubusercontent.com/4503659/209703204-d906ec82-faa8-40ae-a407-28cf44c87053.png)

**Comment by schmanu** [comment]
In my opinion our tx service should remove all confirmations of the wallet address when an owner gets removed or at least not return those confirmations anymore.

What do you think @Uxio0?

In the meantime I will investigate why we should 1 of 2 in the summary and 2 of 2 in the confirmations list 🤔

**Comment by schmanu** [comment]
@francovenica I cannot reproduce this behavior. For me the confirmations disappear correctly.

Could it be caused by  #1374? Could you hit refresh once the owner removal is indexed and your transaction summary shows "1 of 2"?

**Comment by francovenica** [comment]
@schmanu 
I tried again and the issue of the signatures being "1 out of 2" and confirmations being "2 out of 2" still shows. 
If you see the tx in its "deep link view" (the one using the share icon) then it still shows 2 out of 2 signatures for some reason.
Still the main issue remains. If a 3rd owner that wants to sign the tx so it can be finally executed will see the error message GS026 I mentioned in a previous comment.

Check this tx here, and then in the queue tab. 
Try to execute it with a non-owner and you will see the GS error
https://app.safe.global/gor:0x24F8542Da0C5f267c177a224270442C73804EE80/transactions/tx?id=multisig_0x24F8542Da0C5f267c177a224270442C73804EE80_0xcade98b0a87c66670c3f2d99493bde9210896c6a071e53ecc9a014ba1dfcb2bf

**Comment by francovenica** [comment]
Snapshot for the previous comment
Queue view
![image](https://user-images.githubusercontent.com/4503659/209938699-7aac3bf9-489c-4e06-a05b-a803183a6973.png)

single tx view:
![image](https://user-images.githubusercontent.com/4503659/209938664-56e09043-b952-40ba-b23e-de0cb77ce59c.png)

**Comment by DiogoSoaress** [comment]
> I tried again and the issue of the signatures being "1 out of 2" and confirmations being "2 out of 2" still shows.

I've opened a PR for that @francovenica -> https://github.com/safe-global/web-core/pull/1468

**Comment by francovenica** [comment]
Thanks @DiogoSoaress . I saw that while testing this so I'll be able to check your fix when's done.

Regarding not being able to sign with a 3rd owner: With @schmanu we reached to the conclusion that it might be a backend cache that "doesn't relaize" that the owner was removed and his signature is no longer valid. After a while the tx updates itself and sets the proper 1out of 2 confirmation and removes the now ex-owner from the "signers list".

I think we can close this ticket as issue that frontend cannot solve, and we could check with backend about this cache
